### ISSUE TITLE
Cannot run generate after "npm run clean" of pristine clone of 3.3.1

### ISSUE THREAD

**Issue by jacobk** [mxstbr/react-boilerplate#1189]
`npm run generate` throws an error when run after `npm run clean`

**My environment**
```shell
</tmp/react-boilerplate> (master)
$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.11.5
BuildVersion:	15F34
</tmp/react-boilerplate> (master)
$ npm --version
3.10.8
</tmp/react-boilerplate> (master)
$ node --version
v5.0.0
</tmp/react-boilerplate> (master=)
$ yarn --version
0.16.1
```

**How to reproduce**
1. Clone 3.3.1
2. Run `npm run setup`
3. Run `npm run clean`
4. Run `npm run generate`
5. Error

**Reproduce error log**
```shell
</tmp>
$ git clone --depth=1 https://github.com/mxstbr/react-boilerplate.git
Cloning into 'react-boilerplate'...
remote: Counting objects: 328, done.
remote: Compressing objects: 100% (280/280), done.
remote: Total 328 (delta 49), reused 201 (delta 33), pack-reused 0
Receiving objects: 100% (328/328), 1.80 MiB | 1.60 MiB/s, done.
Resolving deltas: 100% (49/49), done.
Checking connectivity... done.
</tmp>
$ cd react-boilerplate/
</tmp/react-boilerplate> (master=)
$ git describe --abbrev=0 --tags
v3.3.1
</tmp/react-boilerplate> (master=)
$ npm run setup

> react-boilerplate@3.3.1 presetup /private/tmp/react-boilerplate
> npm i chalk shelljs

react-boilerplate@3.3.1 /private/tmp/react-boilerplate
├─┬ chalk@1.1.3
│ ├── ansi-styles@2.2.1
│ ├── escape-string-regexp@1.0.5
│ ├─┬ has-ansi@2.0.0
│ │ └── ansi-regex@2.0.0
│ ├── strip-ansi@3.0.1
│ └── supports-color@2.0.0
└─┬ shelljs@0.7.5
  ├─┬ glob@7.1.1
  │ ├── fs.realpath@1.0.0
  │ ├─┬ inflight@1.0.6
  │ │ └── wrappy@1.0.2
  │ ├── inherits@2.0.3
  │ ├─┬ minimatch@3.0.3
  │ │ └─┬ brace-expansion@1.1.6
  │ │   ├── balanced-match@0.4.2
  │ │   └── concat-map@0.0.1
  │ ├── once@1.4.0
  │ └── path-is-absolute@1.0.1
  ├── interpret@1.0.1
  └─┬ rechoir@0.6.2
    └── resolve@1.1.7


> react-boilerplate@3.3.1 setup /private/tmp/react-boilerplate
> node ./internals/scripts/setup.js


Cleaning old repository ✓
Installing dependencies ✓. (This might take a while)Installing dependencies
Initialising new repository ✓
Done!
> react-boilerplate@3.3.1 postsetup /private/tmp/react-boilerplate
> npm run build:dll


> react-boilerplate@3.3.1 build:dll /private/tmp/react-boilerplate
> node ./internals/scripts/dependencies.js

Building the Webpack DLL...
Hash: 063867de18d261757760
Version: webpack 2.1.0-beta.25
Time: 4589ms
                      Asset     Size  Chunks             Chunk Names
reactBoilerplateDeps.dll.js  3.01 MB       0  [emitted]  reactBoilerplateDeps
chunk    {0} reactBoilerplateDeps.dll.js (reactBoilerplateDeps) 2.52 MB [entry] [rendered]
    [2] dll reactBoilerplateDeps 12 bytes {0} [built]
     + 681 hidden modules
</tmp/react-boilerplate> (master)
$ npm run clean

> react-boilerplate@3.3.1 clean /private/tmp/react-boilerplate
> shjs ./internals/scripts/clean.js

Cleanup started... ✓
Cleanup done. Happy Coding!!!
</tmp/react-boilerplate> (master)
$ npm run generate

> react-boilerplate@3.3.1 generate /private/tmp/react-boilerplate
> plop --plopfile internals/generators/index.js

/private/tmp/react-boilerplate/internals/generators/component/index.js:35
    let componentTemplate;
    ^^^

SyntaxError: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:404:25)
    at Object.Module._extensions..js (module.js:432:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:311:12)
    at Module.require (module.js:366:17)
    at require (module.js:385:17)
    at Object.<anonymous> (/private/tmp/react-boilerplate/internals/generators/index.js:8:28)
    at Module._compile (module.js:425:26)
    at Object.Module._extensions..js (module.js:432:10)

npm ERR! Darwin 15.5.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "generate"
npm ERR! node v5.0.0
npm ERR! npm  v3.10.8
npm ERR! code ELIFECYCLE
npm ERR! react-boilerplate@3.3.1 generate: `plop --plopfile internals/generators/index.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the react-boilerplate@3.3.1 generate script 'plop --plopfile internals/generators/index.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the react-boilerplate package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     plop --plopfile internals/generators/index.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs react-boilerplate
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls react-boilerplate
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /private/tmp/react-boilerplate/npm-debug.log
</tmp/react-boilerplate> (master)
```

**Comment by jacobk** [comment]
I just noticed that running `npm run generate` on a non cleaned freshly `setup`ed version causes the same issue.

**Comment by jacobk** [comment]
Upgrading node version (to 7.0.0) solved it.

**Comment by samit4me** [comment]
Thank you for reporting the issue, the level of detail is amazing, really appreciated 👌  

I've done some investigations myself and I can confirm this issue with `node v5.0.0`. According to [Node Green](http://node.green/#let) Node v5 should support the use of `let`, but it appears that this is not true given the error message:
`Block-scoped declarations (let, const, function, class) not yet supported outside strict mode`. 

***I can confirm Node v6+ work fine on macOS with or without `yarn`.***

### Now is this a bug? Good question! 
According to the [FAQs](https://github.com/mxstbr/react-boilerplate/blob/master/docs/general/faq.md#im-using-node-v012-and-the-server-doesnt-work), this boilerplate supports the last three major versions of Node.js, so that would be v5, v6, and v7. Also given that Node v6.9.0 is now (LTS) I would say that Node v4 is no longer supported by this boilerplate, what do you think?

*FYI: if we add "use strict" to the top of `internals/generators/component/index.js` or change the `let` to a `var` the command `npm run generate` does work with Node v5.*

### Windows issues with `yarn`!
During my testing with *(Windows 10, Node v6.6.0)* without `yarn` the command `npm run generate` worked perfectly. After installing `yarn`, it broke when running `npm run setup`, which looks like it might be a related to this [open issue with Yarn](https://github.com/yarnpkg/yarn/issues/1619)! After looking at the `yarn-error.log`, this was the error message:

```
Trace: 
  Error: https://registry.yarnpkg.com/through/-/through-2.3.8.tgz: invalid tar file
      at Extract.Parse._startEntry (C:\Users\sam\AppData\Roaming\npm\node_modules\yarn\node_modules\tar\lib\parse.js:149:13)
      at Extract.Parse._process (C:\Users\sam\AppData\Roaming\npm\node_modules\yarn\node_modules\tar\lib\parse.js:131:12)
      at BlockStream.<anonymous> (C:\Users\sam\AppData\Roaming\npm\node_modules\yarn\node_modules\tar\lib\parse.js:47:8)
      at emitOne (events.js:96:13)
      at BlockStream.emit (events.js:188:7)
      at BlockStream._emitChunk (C:\Users\sam\AppData\Roaming\npm\node_modules\yarn\node_modules\block-stream\block-stream.js:145:10)
      at BlockStream.flush (C:\Users\sam\AppData\Roaming\npm\node_modules\yarn\node_modules\block-stream\block-stream.js:70:8)
      at BlockStream.end (C:\Users\sam\AppData\Roaming\npm\node_modules\yarn\node_modules\block-stream\block-stream.js:66:8)
      at Extract.Parse.end (C:\Users\sam\AppData\Roaming\npm\node_modules\yarn\node_modules\tar\lib\parse.js:86:23)
      at UnpackStream.onend (_stream_readable.js:511:10)
```

Interestingly tho, after installing Node v7.0.0 on Windows 10 it worked perfectly and `npm run generate` worked as intended. So, as much as I dislike handballing issues, this might be better followed up in the [open issue with `yarn`](https://github.com/yarnpkg/yarn/issues/1619).

Important to know that each time I changed a version of Node or tried with or without `yarn` on either Windows or macOS, I removed react-boilerplate and cloned it again etc, so yes this took some time :)

**Comment by samit4me** [comment]
Okay, after further testing it seems that the issues with `Windows + yarn + react-boilerplate` has disappeared. I'm not sure exactly what fixed it but I did run `yarn cache clean` and I also removed `yarn` and installed it again (v0.16.1). I've now tried Windows 10 with Node v5.0.0, v6.6.0, v6.8.1, v6.9.1, and v7.0.0 and running `npm run setup`, 'npm run clean`, and `npm run generate` all work perfectly with each version of node.
